libs/live-nft-react/src/hooks/useNftGalleryFilter.ts

```diff
@@ -0,0 +1,71 @@
+import { useMemo } from "react";
+import { InfiniteData, useInfiniteQuery, UseInfiniteQueryResult } from "@tanstack/react-query";
+import { fetchNftsFromSimpleHash } from "@ledgerhq/live-nft/api/index";
+import { SimpleHashResponse } from "@ledgerhq/live-nft/api/types";
+import { ProtoNFT } from "@ledgerhq/types-live";
+
+export type HookProps = {
+  addresses: string;
+  nftsOwned: ProtoNFT[];
+  chains: string[];
+};
+
+export type PartialProtoNFT = Partial<ProtoNFT>;
+
+export type NftGalleryFilterResult = UseInfiniteQueryResult<
+  InfiniteData<SimpleHashResponse, unknown>,
+  Error
+> & {
+  nfts: ProtoNFT[];
+};
+
+/**
+ * useNftGalleryFilter() will apply a spam filtering on top of existing NFT data.
+ * - addresses: a list of wallet addresses separated by a ","
+ * - nftOwned: the array of all nfts as found by all user's account on Ledger Live
+ * - chains: a list of selected network to search for NFTs
+ * NB: for performance, make sure that addresses, nftOwned and chains are memoized
+ */
+export function useNftGalleryFilter({
+  addresses,
+  nftsOwned,
+  chains,
+}: HookProps): NftGalleryFilterResult {
+  const queryResult = useInfiniteQuery({
```

Comment:
on separation of concerns we have splitted the logic out of the react part already, so  if an integration would want to use another way and not react-query, they could directly consume the live-nft lib and not the -react specifics' (the separation is in this library)

hiding the fact it's react-query but having the same api as react-query feels it would become an overhead. today we can directly use that lib and benefit the types.

with a `useInfiniteQuery` that "is not react-query but it's the same API", we need to do a adaptor to the same props it takes and this adaptor instantly become a tech debt.

> what happens if tomorrow we want to move from react-query to another library?

if tomorrow we switch to another library we refactor this hook 🤔  the same way we would have to refactor the adaptor (that today we don't need to impleemnt)


---

side note: if the goal is more to reconsider the library, we always can and i think in that direction we probably could make that hook "fetch it all" at once, because ultimately the pagination is a bit overkill for most users. but if we need a pagination system, we would either need to figure out a common pattern to type what is a "paginating hook" but i think it shouldn't be the exact same `useInfiniteQuery` type, but a generic version of this idea